analyzere_extras/visualizations.py

```diff
@@ -0,0 +1,383 @@
+import hashlib
+import itertools
+from collections import OrderedDict
+
+from analyzere import LayerView
+from graphviz import Digraph
+try:
+    from IPython.display import FileLink
+    _in_notebook = True
+except ImportError:
+    _in_notebook = False
+
+from sys import float_info
+
+
+def _format_description(description):
+    """Clean the description of a node for display in Graphviz"""
+    return description.replace('\'', '').encode('unicode_escape').decode()
+
+
+def _format_DateField(df):
+    """Format a Date field, chopping off the time portion"""
+    return df.date().isoformat()
+
+
+def _format_MoneyField(mf):
+    """Format a MoneyField to '<unlimited|value> CCY' """
+    if mf.value == float_info.max:
+        formatted = 'unlimited'
+    else:
+        formatted = '{:,.0f} {}'.format(float(mf.value), mf.currency)
+    return formatted
+
+
+def _format_filters(filters):
+    num_filters = len(filters)
+    if num_filters == 0:
+        filters_value = '(empty)'
+    elif num_filters < 4:
+        filter_strings = []
+        for f in filters:
+            filter_strings.append("'{}'".format(f.name))
+        filters_value = '[{}]'.format(', '.join(filter_strings))
+    else:
+        filters_value = '({} filters)'.format(num_filters)
+    return filters_value
+
+
+def _format_reinstatements(reinstatements):
+    num_reinstatements = len(reinstatements)
+
+    if num_reinstatements > 4:
+        reinsts_value = '{}'.format(num_reinstatements)
+    else:
+        reinstatements = ['{}/{}'.format(r.premium, r.brokerage)
+                          for r in reinstatements]
+        reinsts_value = '[{}]'.format(', '.join(reinstatements))
+    return reinsts_value
+
+
+def _format_coverage(layer):
+    coverage = ['{}'.format(_format_DateField(layer.inception_date))
+                if hasattr(layer, 'inception_date') else '-inf',
+                '{}'.format(_format_DateField(layer.expiry_date))
+                if hasattr(layer, 'expiry_date') else 'inf']
+    return '[{}]'.format(', '.join(coverage))
+
+
+class FormattingHelper:
+    def __init__(self, layer):
+        self._layer = layer
+        self._terms = OrderedDict()
+        self.warning = False
+
+    def append(self, required_attr, attr_formatted=None,
```

Comment:
@bmacphee I'd rather see this called something like `display_name` personally, but that's preference.